File: api/filing/models/bank_account_model.py
from .api_base_model import ApiBaseModel


class BankAccountModel(ApiBaseModel):
    bank_account_id: int | None = None
    person_id: int | None = None
    account_number: str
    ifsc_code: str
    bank_name: str
    account_type: str
    is_primary: bool = False

File: web/filing/models/person/bank-account-model.ts
/**
 * Bank Account Model
 * Mirrors: bank_account in FilingModel
 */

export interface BankAccountModel {
  bankAccountId?: number | null
  personId?: number | null
  accountNumber: string
  ifscCode: string
  bankName: string
  accountType: string
  isPrimary: boolean
}

// ==================== Widget Types ====================

/**
 * Bank Account API Response
 */
export interface BankAccountResponse {
  bankAccount: any
  message?: string
}
